### ISSUE TITLE
output paramater is ignored, original images are overwritten

### ISSUE THREAD

**Issue by currentrame** [imagemin/imagemin#163]
imagemin v5.0.0
imagemin-pngquant v5.0.0

code:
`const imagemin = require('imagemin');
    var pngComp = require("imagemin-pngquant");
    imagemin(['/Users/m/projects/m/_test/tmp/in/*.{jpg,png}'], '/Users/m/projects/m/_test/tmp/out', {
        use: [
            pngComp({quality: '65-80'})
        ]
    }).then(files => {
        console.log(files);
    });`

**Comment by kevva** [comment]
If you log `files`, what does it output?

**Comment by QuentinFchx** [comment]
https://github.com/imagemin/imagemin/blob/master/index.js#L11
`const dest = output ? path.resolve(output, input) : null;`
Not sure how this can work if `input` is already a relative path to the file ?

**Comment by kevva** [comment]
@QuentinFchx, happy if you could add a failing test.

**Comment by QuentinFchx** [comment]
```js
test('should output at the specified location', async t => {
	const files1 = await m(['fixture.jpg'], 'output', {use: imageminJpegtran()});
	const files2 = await m(['output/*.jpg'], 'output', {use: imageminJpegtran()});
	t.is(path.relative(__dirname, files1[0].path), 'output/fixture.jpg');
	t.is(path.relative(__dirname, files2[0].path), 'output/fixture.jpg'); // will fail
});
```

**Comment by ldanet** [comment]
I'm having the same issue.
As QuentinFchx pointed out, there is a problem with globs, but there is another problem. currentrame and I are using absolute paths, and `path.resolve(output, input)` will simply return `input` if it is an absolute path. As a result, `output` gets completely ignored.
My app's working directory, input path and output path are completely unrelated, so relative paths don't make sense in my case.

**Comment by QuentinFchx** [comment]
We might want to use https://github.com/contra/glob2base.

**Comment by kevva** [comment]
Will try it and see what I come up with.

**Comment by kevva** [comment]
@QuentinFchx @currentrame @ldanet, I think #175 should fix it.

**Comment by QuentinFchx** [comment]
@kevva It works for simple patterns, but when you're using globstars (`**/*.jpg`) it does not

**Comment by kevva** [comment]
Isn't [this](https://github.com/imagemin/imagemin/blob/master/test.js#L57-L70) exactly what you wrote earlier?

**Comment by QuentinFchx** [comment]
I wrote a specific test case that is fixed by your commit, however I forgot to mention this case can be extended.

Let's take the following use case:

I want to process`bar.jpg` located at `images/foo/bar.jpg`. I want the output to be send into the `ouput` folder **with the same directory** structure `output/foo/bar.jpg`.
According to the current implementation, and if I'm not mistaken:
- I call `imagemin(['images/**/*.jpg'], 'output')`
- The `globby` lib parse the globs and returns images paths (here `images/foo/bar.jpg`)
- The `handleFile` function is called with `handleFile('images/foo/bar.jpg', 'output')
- The file is output at `output/bar.jpg` instead of `output/foo/bar.jpg`

TLDR; The issue is that we lose the reference of the "matching" part of the glob. For each file, we want to separate the "static" part of the glob (`/images`) from the "dynamic" part `/**/*.jpg`, and append that "dynamic" part to the "output" (`/output`) one if provided.

Prior to 5.0.0 version, we were using [vinyl-fs](https://github.com/gulpjs/vinyl-fs) that used to keep that "matching pattern". It does not seem possible to achieve with globby alone.

**Comment by kevva** [comment]
Do we want that as the default behavior or behind an option? We only need to do this in any case https://github.com/sindresorhus/cpy/blob/master/index.js#L36.

**Comment by davidchase** [comment]
I think i may have a solution for this issue with [path-union](https://github.com/davidchase/path-union)

**Comment by davidchase** [comment]
I think i may have a solution with [path-union](https://github.com/davidchase/path-union)

updated `index.js`
```diff
-const dest = output ? path.join(output, path.basename(input)) : null;
+const dest = output ? union(path.resolve(output), path.resolve(input)) : null;
```

the command:

```js
im(['./images/**/*.jpg'], 'output')
```

my directory for images:

```sh
λ find images
images
images/fixture-corrupt.jpg
images/test
images/test/fixture.jpg
```

and output 

```sh
λ find output
output
output/images
output/images/fixture-corrupt.jpg
output/images/test
output/images/test/fixture.jpg
```

if this is something of interest to use.. i can update tests and send a PR

**Comment by davidchase** [comment]
I think my [path-union](https://github.com/davidchase/path-union) does the trick:

```diff
- const dest = output ? path.join(output, path.basename(input)) : null;
+ const dest = output ? union(path.resolve(output), path.resolve(input)) : null;
```

running `imagemin(['images/**/*.jpg'], 'output')`
```sh
λ find images
images
images/fixture-corrupt.jpg
images/test
images/test/fixture.jpg
```

```sh
λ find output
output
output/images
output/images/fixture-corrupt.jpg
output/images/test
output/images/test/fixture.jpg
```
